nameserver/entry.go

```diff
@@ -233,8 +233,8 @@ type GossipData struct {
 }
 
 func (g *GossipData) Merge(o router.GossipData) {
-	checkAndPanic(CaseSensitive(g.Entries))
-	defer func() { checkAndPanic(CaseSensitive(g.Entries)) }()
+	checkAndPanic(CaseInsensitive(g.Entries))
+	defer func() { checkAndPanic(CaseInsensitive(g.Entries)) }()
```

Comment:
Just clarifying for my own understanding: `GossipData.Merge` didn't need to check `g.Entries` because `Entries.merge()` already did that.  But `Entries.merge()` also relies on its `incoming` argument being sorted, so it should check that.

Note that `Entries.merge()` is called in two ways: for internally-generated outgoing data and for incoming gossip from another peer.  In the latter case we re-sort the incoming data so it should pass the test. We have traditionally preferred to raise an error and drop the connection when bad data is received from another peer, as opposed to panicking.